### PR TITLE
Extra error checks for K0 and C-Phi

### PR DESCRIPTION
**Pull request by WPK4FEM**
**📝 Description**
Extra errors and unit tests for those checks for K0 and C-Phi

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp [hunk 1]
```diff
@@ -26,6 +26,8 @@ void ApplyCPhiReductionProcess::ExecuteInitializeSolutionStep()
 
     if (IsStepRestarted()) mReductionIncrement *= 0.5;
     mReductionFactor = mPreviousReductionFactor - mReductionIncrement;
+    KRATOS_ERROR_IF(mReductionFactor <= 0.01)
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp:29]: As far as I understand this is an extra stop criteria. What is your opinion it is useful to allow the end user to provide this value?

File: applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp [hunk 2]
```diff
@@ -64,7 +66,14 @@ void ApplyCPhiReductionProcess::ExecuteFinalize()
     KRATOS_INFO("ApplyCPhiReductionProcess") << "Final safety factor = " << 1.0 / mReductionFactor << std::endl;
 }
 
-double ApplyCPhiReductionProcess::GetAndCheckPhi(const Element::PropertiesType& rProp)
+int ApplyCPhiReductionProcess::Check()
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp:69]: Something is strange with git on my laptop and I cannot switch to your branch. Please could you explain where Check function is called? Thank you

**Review by markelov208 — DISMISSED** [review]
Hi Wijtze-Pieter, it looks like the users are heavily use this functionality. Thank you very much for adding new checks that are unit tested. I have two non-blocking comments.

File: applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp (hunk 1 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp:29]: Indeed this is a stop criterion. Reducing below zero is non-physical, near zero the soil behaves like a liquid. I didn't want to introduce another parameter, so I choose 1 %. I will write that down in the documentation.

File: applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp (hunk 2 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp:69]: The comment for this in the process base class is:
```c++
/* @brief This function is designed for being called after ExecuteInitialize ONCE
     * to verify that the input is correct.
     * /
```

File: applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp (hunk 1 above)
Comment (markelov208) [reply applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.cpp:29]: Thank you for the clarification

**Review by avdg81 — APPROVED** [review]
This looks more robust to me now. I have no further comments or suggestions. Let's merge it when it's ready.

**Review by markelov208 — APPROVED** [review]

**Pull request merged**